nshlib/nsh_syscmds.c

```diff
@@ -518,3 +518,21 @@ int cmd_uname(FAR struct nsh_vtbl_s *vtbl, int argc, char **argv)
   return OK;
 }
 #endif
+
+#if defined(CONFIG_BOARDCTL_RESET_CAUSE) && !defined(CONFIG_NSH_DISABLE_RESET_CAUSE)
+int cmd_reset_cause(FAR struct nsh_vtbl_s *vtbl, int argc, char **argv)
+{
+  int ret;
+  struct boardioc_reset_cause_s cause;
+  memset(&cause, 0, sizeof(cause));
+
+  ret = boardctl(BOARDIOC_RESET_CAUSE, &cause);
+  if (ret < 0)
+    {
+       nsh_error(vtbl, g_fmtcmdfailed, argv[0], "boardctl", NSH_ERRNO);
+       return ERROR;
+    }
+  nsh_output(vtbl, "cause:%x, flag:%x\n", cause.cause, cause.flag);
```

Comment:
should we map cause to string xxx(yy)?

Reply:
My intention is: this is just a helper cmd to test whether reset cause works.
It may be useful in developing stage, And a developer should know those causes and flags.
I don't want to increase .rodata size or code size for this helper cmd.

Reply:
I'd prefer to make it as thin as possible.